### DISCUSSION TITLE
Possible to nest uno.css under a rule?

### DISCUSSION THREAD

**Discussion by carlosingles** [unocss/unocss#1140] [Q&A] [answered]
Was just wondering if it's possible to nest the `import uno.css` global mode, beneath a css rule.

i.e.

```css
.some-rule {
  /* uno.css rules go here */
}
```

**Comment by chu121su12** [comment]
It's not in the readme, but you could try using scope 

https://github.com/unocss/unocss/blob/979b6936c2c47e0c989076c7d259d6276fc09c95/test/scope.test.ts#L26-L29

https://github.com/unocss/unocss/blob/979b6936c2c47e0c989076c7d259d6276fc09c95/test/__snapshots__/scope.test.ts.snap#L3-L5

**Comment by carlosingles** [comment, marked as answer]
Thanks for the response!

I looked into this option, however I was using uno.css with vite within a vue project and did not want to seperate the css from the vite configuration.

I ended up using variants to achieve my desired result:

```ts
// vite.config.ts
import Unocss from "unocss/vite";

export default {
  plugins: [
    Unocss({
      variants: [
        (matcher) => {
          return {
            matcher,
            selector: (s) => `.some-rule ${s}`,
          };
        },
      ],
    }),
  ],
};
```

Which allowed me to continue using the vite configuration with the prefix before every utility class.

If there are things I should consider when using this approach please let me know 😄
